src/pages/index.js

```diff
@@ -129,26 +194,88 @@ function closeModalOnEsc(evt) {
 // Allows for editing profile section by submitting info to the connected form
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  evt.target.reset();
-  closeModal(editModal);
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      evt.target.reset();
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+    });
 }
 
 // Allows for creating a new card/new post by submitting info to the connected form
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
-  console.log(cardCaptionInput.value);
-  console.log(cardLinkInput.value);
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
   const inputValues = {
     name: cardCaptionInput.value,
     link: cardLinkInput.value,
   };
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-  evt.target.reset();
-  disableButton(cardSubmitButton, settings);
-  closeModal(cardModal);
+  api
+    .addNewCards(inputValues)
+    .then((cardData) => {
+      const cardElement = getCardElement(cardData);
+      cardsList.prepend(cardElement);
+      evt.target.reset();
+      disableButton(cardSubmitButton, settings);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+      closeModal(cardModal);
+    });
+}
+
+// Allows for swapping out a profile avatar image
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+  api
+    .editAvatar(avatarInput.value)
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+      evt.target.reset();
```

Comment:
* Please, **disable** the `avatar` submit button after submission as well https://snipboard.io/7xcyMB.jpg 


```js
      evt.target.reset();
      disableButton(submitBtn, settings);
```